NOTICE.txt

```diff
@@ -0,0 +1,13 @@
+Apache Arrow Java
+Copyright 2016-2024 The Apache Software Foundation
+
+This product includes software developed at
+The Apache Software Foundation (http://www.apache.org/).
+
+This product includes software from Dremio (Apache 2.0)
+  * Copyright (C) 2017-2018 Dremio Corporation
+  * https://github.com/dremio/dremio-oss
+
+This product includes software from Google Guava (Apache 2.0)
+  * Copyright (C) 2007 The Guava Authors
+  * https://github.com/google/guava
```

Comment:
Can we move this to `LICENSE.txt`?

https://infra.apache.org/licensing-howto.html#alv2-dep

> Bundling an Apache 2-0-licensed dependency
>
> Assuming that the bundled dependency itself contains no bundled sub-components under other licenses, so the ALv2 applies uniformly to all files, there is no need to modify LICENSE. However, for completeness it is useful to list the products and their versions, as is done for products under other licenses.
>
> If the dependency supplies a NOTICE file, its contents must be analyzed and the relevant portions bubbled up into the top-level NOTICE file.

Reply:
I am ok moving it to `LICENSE.txt` but to be honest I am unsure I understand what the `NOTICE.txt` refers then when it says, here https://infra.apache.org/licensing-howto.html#mod-notice:
```
Copyright notifications which have been relocated, rather than removed, from source files must be preserved in NOTICE. However, elements such as the copyright notifications embedded within BSD and MIT licenses do not need to be duplicated in NOTICE. You can leave those notices in their original locations.
```
or here: https://www.apache.org/legal/src-headers.html#notice
```
Use the remainder of the NOTICE file to include [required third-party notices][1]. The NOTICE file may also include copyright notices moved from source files submitted to the ASF.
```